src/libmongoc/src/mongoc/mongoc-collection.h

```diff
@@ -351,6 +351,30 @@ mongoc_collection_estimated_document_count (
    bson_t *reply,
    bson_error_t *error);
 
+
+
+
+typedef struct _mongoc_index_model_t mongoc_index_model_t;
+typedef struct _mongoc_index_options_t mongoc_index_options_t;
```

Comment:
It's a bit annoying that this name is so similar to the deprecated `mongoc_index_opts_t`.  I do like having a type like this though, as long as it is opaque.